src/odbc_connection.cpp

```diff
@@ -1271,16 +1271,30 @@ class CallProcedureAsyncWorker : public ODBCAsyncWorker {
 
       SQLRETURN return_code;
 
-      char *combinedProcedureName = new char[255]();
-      if (data->catalog != NULL) {
-        strcat(combinedProcedureName, (const char*)data->catalog);
-        strcat(combinedProcedureName, ".");
-      }
-      if (data->schema != NULL) {
-        strcat(combinedProcedureName, (const char*)data->schema);
-        strcat(combinedProcedureName, ".");
-      }
-      strcat(combinedProcedureName, (const char*)data->procedure);
+      #ifndef UNICODE
+      char *combinedProcedureName = new char[1024]();
+      sprintf (
+        combinedProcedureName,
+        "%s%s%s%s%s",
+        data->catalog ? data->catalog : (SQLCHAR *)"",
+        data->catalog ? "." : (const char*)"",
+        data->schema ? data->schema : (SQLCHAR *)"",
+        data->schema ? "." : (const char*)"",
```

Comment:
```suggestion
        data->catalog ? (const char*) data->catalog : "",
        data->catalog ? "." : "",
        data->schema ? (const char*) data->schema : "",
        data->schema ? "." : "",
```

Casting `""` to a `const char*` is pointless since it already is that. Casting a `""` to a `SQLCHAR*` is "dangerous", since you're casting away the `const`. In this case it's fine, but since `%s` expects a `const char*` anyway (and GCC will warn at different warning levels), it's better to cast the `SQLCHAR*` to `const char*`.

Reply:
my GCC was throwing errors without the cast, saying that it was a `void *` and that `"."` and `""` had to be of the same type.

Reply:
I must have read the error wrong. Was probably upset that I was comparing SQLCHAR * to void*/const char*. With your changes it compiles now.

Reply:
yeah, both sides of a ternary need to be the same type. You're fix was mostly correct, just better to fix the other side of the ternary :)